### ISSUE TITLE
Support embedded/referenced object validation

### ISSUE THREAD

**Issue by mavarazy** [RxNT/json-rules-engine-simplified#12]
Currently this objects are ignored by the validator

**Comment by CodestarGames** [comment]
Will this allow for referencing other nested properties in the form object? I'm currently trying to figure out how to compare two properties I have on an object that I have a schema for. 

`var rules = [{
    conditions: {
        cost: {
            decreasedByMoreThanPercent: {
                average: "averages_monthly.cost",
                target: 20
            }
        }
    },
    event: {
        type: "remove",
        params: {
            field: "password"
        },
    }
}];

var formData = {
    cost: 50,
    averages_monthly: {
        cost: 100
    }
}
`

**Comment by mavarazy** [comment]
@CodestarGames As I understand you want to have a condition looking something like
```js
 conditions: {
     cost: {
        decreasedByMoreThanPercent: {
             average: "averages_monthly.cost",
             target: "cost"
         }
     }
 }
```
This specific issue, is not for that. This is for validating references like "averages_monthly.cost", that averages_monthly actually has a cost field. This can be a good feature, I'll open a separate issue to address this.

**Comment by mavarazy** [comment]
@CodestarGames I've fixed #14, so now it should work, by appending `$` to the field definition, like this.
```js
 conditions: {
     cost: {
        decreasedByMoreThanPercent: {
             average: "$averages_monthly.cost",
             target: 20
         }
     }
 }
```
The fix is in the latest release 0.1.11
